File: src/main/java/com/ultreon/bubbles/init/TextureCollections.java
package com.ultreon.bubbles.init;

import com.ultreon.bubbles.InternalMod;
import com.ultreon.bubbles.common.mod.ModInstance;
import com.ultreon.bubbles.event.bus.ModEvents;
import com.ultreon.bubbles.registry.Registers;
import com.ultreon.hydro.event.bus.GameEvents;
import com.ultreon.hydro.registry.DeferredRegister;
import com.ultreon.hydro.registry.object.RegistryObject;
import com.ultreon.hydro.render.TextureCollection;

import java.util.function.Supplier;

/**
 * Initialization for texture collections.
 *
 * @see Registers#TEXTURE_COLLECTIONS
 * @since 1.0.924-a1
 */
public class TextureCollections {
    private static final DeferredRegister<TextureCollection> TEXTURE_COLLECTIONS = DeferredRegister.create(InternalMod.MOD_ID, Registers.TEXTURE_COLLECTIONS);

    public static final RegistryObject<TextureCollection> BUBBLE_TEXTURES = register("bubble", TextureCollection::new);

    @SuppressWarnings("SameParameterValue")
    private static <T extends TextureCollection> RegistryObject<T> register(String name, Supplier<T> supplier) {
        return TEXTURE_COLLECTIONS.register(name, supplier);
    }

    public static void register(GameEvents eventBus) {
        TEXTURE_COLLECTIONS.register(eventBus);
    }
}

File: src/main/kotlin/qtech/bubbles/init/TextureCollections.kt
package qtech.bubbles.init

import qtech.bubbles.BBInternalMod
import qtech.bubbles.common.mods.ModInstance
import qtech.bubbles.event.bus.LocalAddonEventBus
import qtech.hydro.TextureCollection
import qtech.bubbles.registry.DeferredRegister
import qtech.bubbles.registry.Registers
import qtech.bubbles.registry.`object`.RegistryObject
import java.util.function.Supplier

/**
 * Initialization for texture collections.
 *
 * @see Registers.TEXTURE_COLLECTIONS
 *
 * @since 1.0.924-a1
 */
object TextureCollections {
    private val TEXTURE_COLLECTIONS: DeferredRegister<TextureCollection> = DeferredRegister.create(BBInternalMod.ADDON_ID, Registers.TEXTURE_COLLECTIONS)
    @JvmField
    val BUBBLE_TEXTURES: RegistryObject<TextureCollection> = register("bubble") { TextureCollection() }
    private fun register(name: String, supplier: Supplier<TextureCollection>): RegistryObject<TextureCollection> {
        return TEXTURE_COLLECTIONS.register(name, supplier)
    }

    @JvmStatic
    fun register(eventBus: LocalAddonEventBus<out ModInstance>) {
        TEXTURE_COLLECTIONS.register(eventBus)
    }
}